src/data/types.ts

```diff
@@ -32,4 +32,14 @@ export const PineconeRecordSchema = Type.Object(
 export type RecordId = Static<typeof RecordIdSchema>;
 export type Values = Static<typeof RecordValuesSchema>;
 export type SparseValues = Static<typeof RecordSparseValuesSchema>;
-export type PineconeRecord = Static<typeof PineconeRecordSchema>;
+export type RecordValues = Array<number>;
+export type RecordSparseValuse = {
+  indices: Array<number>;
+  values: Array<number>;
+};
+export type PineconeRecord<T> = {
```

Comment:
I think we can make this more specific -- our metadata can't be nested and can only have values of string, boolean, number, or array of strings

`PineconeRecord<T extends Record<string, string | boolean | number | array<string>>`

(might want to make a type for `type MetadataValue = string | boolean | number | array<string>`)

I've done some similar stuff for both metadata values and the filter here: https://github.com/pinecone-io/management-console/blob/main/apps/console/src/types/metadata/language.ts#L1

Reply:
Thanks, I'll give this a shot.